demo/wayland_rawfb/nuklear_raw_wayland.h

```diff
@@ -694,21 +694,21 @@ void nk_wayland_draw_text(const struct nk_wayland *win, const struct nk_user_fon
     int glyph_len = 0;
     int next_glyph_len = 0;
     struct nk_user_font_glyph g;
-    if (!len || !text) return;
+    if (!text.ptr || !text.len) return;
 
     x = 0;
-    glyph_len = nk_utf_decode(text, &unicode, len);
+    glyph_len = nk_utf_decode(text, &unicode);
     if (!glyph_len) return;
 
     // draw every glyph image 
-    while (text_len < len && glyph_len) {
+    while (text_len < text.len && glyph_len) {
         struct nk_rect src_rect;
         struct nk_rect dst_rect;
         float char_width = 0;
         if (unicode == NK_UTF_INVALID) break;
 
         // query currently drawn glyph information 
-        next_glyph_len = nk_utf_decode(text + text_len + glyph_len, &next, (int)len - text_len);
+        next_glyph_len = nk_utf_decode(nk_substr(text, text_len + glyph_len, text.len), &next);
```

Comment:
I don't quite grok what's going on here (even after a cursory glance at nk_utf_decode) but they look different to me.

The old decode would get
len-text_len as c_len
while the new version gets
text.len - (text_len+glyph_len) as str.len aka "c_len"

Am I missing something or is the new version giving a shorter input string/slice?  Was the original wrong?

Reply:
Good catch. I don't quite remember why the code ended up like this, so I'll need to look this over again.

Reply:
I remember why I did this now that I look at it. I was looking over the code and was like "nk_utf_decode only decodes one glyph. I'm pretty sure we don't need to shorten the length of the string here. This will make the code simpler". I'm pretty sure I'm correct in making this simplification.